gdal/ogr/ogrsf_frmts/lvbag/ogrlvbaglayer.cpp

```diff
@@ -0,0 +1,444 @@
+            {
+                OGRSpatialReference* poSRS = new OGRSpatialReference{};
+                poSRS->importFromURN(papszIter[1]);
+                poGeomField->SetSpatialRef(poSRS);
+                poSRS->Release();
+            }
+
+            osElementString += " ";
+            osElementString += papszIter[0];
+            osElementString += "=\"";
+            osElementString += papszIter[1];
+            osElementString += "\"";
+            papszIter += 2;
+        }
+
+        osElementString += ">";
+    }
+    else if( nFeatureCollectionDepth > 0 && nFeatureElementDepth == 0 &&
+             EQUAL("sl-bag-extract:bagObject", pszName) )
+    {
+        nFeatureElementDepth = nCurrentDepth;
+        if( !bSchemaOlny )
+            poFeature = new OGRFeature(poFeatureDefn);
+    }
+    else if( nFeatureCollectionDepth == 0 && EQUAL("sl:standBestand", pszName) )
+    {
+        nFeatureCollectionDepth = nCurrentDepth;
+    }
+
+    nCurrentDepth++;
+}
+
+/************************************************************************/
+/*                           endElementCbk()                            */
+/************************************************************************/
+
+void OGRLVBAGLayer::EndElementCbk(const char *pszName)
+{
+    nCurrentDepth--;
+
+    if( nCurrentDepth > nAttributeElementDepth && nAttributeElementDepth > 0
+    && nGeometryElementDepth == 0 )
+    {
+        const char *pszTag = XMLTagSplit(pszName);
+        
+        StopDataCollect();
+        if ( !osElementString.empty() ) 
+        {
+            if( bSchemaOlny )
+            {
+                OGRFieldType eType = OFTString;
+                OGRFieldDefn oField(pszTag, eType);
+                poFeatureDefn->AddFieldDefn(&oField);
+            }
+            else
+            {
+                const int iFieldIndex = poFeatureDefn->GetFieldIndex(pszTag);
+                if( iFieldIndex > -1 )
+                {
+                    poFeature->SetField(iFieldIndex, osElementString.c_str());
+                }
+            }
+        }
+        osElementString.Clear();
+    }
+    else if( nAttributeElementDepth == nCurrentDepth )
+    {
+        nAttributeElementDepth = 0;
+    }
+    else if( nGeometryElementDepth > 0 && nCurrentDepth > nGeometryElementDepth )
+    {
+        osElementString += "</";
+        osElementString += pszName;
+        osElementString += ">";
+    }
+    else if( nGeometryElementDepth == nCurrentDepth )
+    {
+        StopDataCollect();
+        if( !osElementString.empty() )
+        {
+            std::unique_ptr<OGRGeometry> poGeom = std::unique_ptr<OGRGeometry>{
+                CPL_REINTERPRET_CAST(OGRGeometry *, OGR_G_CreateFromGML(osElementString.c_str())) };
+            if( poGeom && !poGeom->IsEmpty() )
+            {
+                if( !bSchemaOlny )
+                    poFeature->SetGeometryDirectly(poGeom.release());
```

Comment:
the SRS should be assigned to poGeom

Reply:
Have a look at the proposed solution. I am not sure about the solution. The JML driver does something akin to the current construction. (63fbf0f)

Reply:
the previous code was OK and should be conserved. I just meant of doing something like poGeom->assignSpatialReference(GetSpatialRef());
Your proposition will have not-so-good performance, and I bet test_ogrsf (cd apps; make test_ogrsf) would complain since it checks that the layer SRS and the SRS of a feature's geometry are the same

Reply:
Like this?

```suggestion
                    poGeom->assignSpatialReference(GetSpatialRef());
                    poFeature->SetGeometryDirectly(poGeom.release());
```

AFAIK the poSRS is only for writable layers.

Reply:
> Like this?

yes